vertical-pod-autoscaler/pkg/utils/vpa/api.go

```diff
@@ -49,14 +49,14 @@ type patchRecord struct {
 	Value interface{} `json:"value"`
 }
 
-func patchVpa(vpaClient vpa_api.VerticalPodAutoscalerInterface, vpaName string, patches []patchRecord) (result *vpa_types.VerticalPodAutoscaler, err error) {
+func patchVpaStatus(vpaClient vpa_api.VerticalPodAutoscalerInterface, vpaName string, patches []patchRecord) (result *vpa_types.VerticalPodAutoscaler, err error) {
 	bytes, err := json.Marshal(patches)
 	if err != nil {
 		klog.Errorf("Cannot marshal VPA status patches %+v. Reason: %+v", patches, err)
 		return
 	}
 
-	return vpaClient.Patch(context.TODO(), vpaName, types.JSONPatchType, bytes, meta.PatchOptions{})
+	return vpaClient.Patch(context.TODO(), vpaName, types.JSONPatchType, bytes, meta.PatchOptions{}, "status")
```

Comment:
Is there a unit test we can add to cover this change?
Maybe using the fake client?

Reply:
It is already have test case on `TestUpdateVpaIfNeeded` https://github.com/kubernetes/autoscaler/blob/1384c8bc09cacd2f774872e01a945c4de6697238/vertical-pod-autoscaler/pkg/utils/vpa/api_test.go#L93-L101

Reply:
Thanks, makes sense.